### PR TITLE
520 adding config for test coverage

### LINKED ISSUE #520: Add test coverage for mocha unit tests

### PR DESCRIPTION
**Pull request by shailee-m**
closes #520 

Added coverage as suggested by meteor docs. This uses the [meteor-coverage](https://github.com/serut/meteor-coverage) package. However, there are some issues with the underlying [babel](https://github.com/istanbuljs/babel-plugin-istanbul) due to which it takes 2 runs of the test to get the correct results

### CODE REVIEW

**Comment by brylie** [comment]
I get a traceback when running the command `meteor` on this branch:

```
 Exception while invoking method 'login' TypeError: Cannot destructure property `isActive` of 'undefined' or 'null'.
I20200305-10:17:41.675(2)?     at lib/accounts.js:44:32
I20200305-10:17:41.675(2)?     at packages/callback-hook/hook.js:130:22
I20200305-10:17:41.675(2)?     at _validateLoginHook.each.callback (packages/accounts-base/accounts_server.js:154:15)
I20200305-10:17:41.675(2)?     at Hook.each (packages/callback-hook/hook.js:108:15)
I20200305-10:17:41.675(2)?     at AccountsServer._validateLogin (packages/accounts-base/accounts_server.js:151:29)
I20200305-10:17:41.675(2)?     at AccountsServer._attemptLogin (packages/accounts-base/accounts_server.js:338:10)
I20200305-10:17:41.676(2)?     at MethodInvocation.methods.login (packages/accounts-base/accounts_server.js:520:23)
I20200305-10:17:41.676(2)?     at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1771:12)
I20200305-10:17:41.676(2)?     at DDP._CurrentMethodInvocation.withValue (packages/ddp-server/livedata_server.js:719:19)
I20200305-10:17:41.676(2)?     at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
I20200305-10:17:41.676(2)?     at DDPServer._CurrentWriteFence.withValue (packages/ddp-server/livedata_server.js:717:46)
I20200305-10:17:41.676(2)?     at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
I20200305-10:17:41.676(2)?     at Promise (packages/ddp-server/livedata_server.js:715:46)
I20200305-10:17:41.676(2)?     at new Promise (<anonymous>)
I20200305-10:17:41.677(2)?     at Session.method (packages/ddp-server/livedata_server.js:689:23)
I20200305-10:17:41.677(2)?     at packages/ddp-server/livedata_server.js:559:43
```

**Comment by brylie** [comment]
The test command runs and I get some failures:

```
I20200305-10:25:09.964(2)?   45 passing (28s)
I20200305-10:25:09.964(2)?   9 failing
I20200305-10:25:09.964(2)? 
I20200305-10:25:09.965(2)?   1) Activities Aggregate Activities Should group by activity name and week  by default:
I20200305-10:25:09.965(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
I20200305-10:25:09.965(2)?   
I20200305-10:25:09.965(2)? 
I20200305-10:25:09.965(2)?   2) Activities Aggregate Activities Should be able to group by activity name and start of month:
I20200305-10:25:09.965(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
I20200305-10:25:09.965(2)?   
I20200305-10:25:09.965(2)? 
I20200305-10:25:09.965(2)?   3) Activities Aggregate Activities Should be able to group by facilitator name and start of week:
I20200305-10:25:09.965(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
I20200305-10:25:09.966(2)?   
I20200305-10:25:09.966(2)? 
I20200305-10:25:09.966(2)?   4) Activities Aggregate Activities Should be able to group by facilitator name and start of month:
I20200305-10:25:09.966(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
I20200305-10:25:09.966(2)?   
I20200305-10:25:09.966(2)? 
I20200305-10:25:09.966(2)?   5) Activities getAggregatedActivities Should group by activity name and week :
I20200305-10:25:09.966(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
I20200305-10:25:09.966(2)?   
I20200305-10:25:09.966(2)? 
I20200305-10:25:09.966(2)?   6) Activities getAggregatedActivities Should be able to group by activity name and start of month:
I20200305-10:25:09.967(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
I20200305-10:25:09.967(2)?   
I20200305-10:25:09.967(2)? 
I20200305-10:25:09.967(2)?   7) Activities getAggregatedActivities Should be able to group by facilitator name and start of week:
I20200305-10:25:09.967(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
I20200305-10:25:09.967(2)?   
I20200305-10:25:09.967(2)? 
I20200305-10:25:09.967(2)?   8) Activities getAggregatedActivities Should be able to group by facilitator name and start of month:
I20200305-10:25:09.967(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
I20200305-10:25:09.967(2)?   
I20200305-10:25:09.967(2)? 
I20200305-10:25:09.968(2)?   9) Activities getAllHomeReportAggregates Should return appropriate data for each aggregation on success:
I20200305-10:25:09.968(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
```

File: README.md [hunk 1]
```diff
@@ -95,6 +95,11 @@ meteor reset will delete the mock database
 This command will start a server on port 8000. It is better to run `npm i` if the tests are run for the first time.
 Server code must be tested on server using `Meteor.isServer` flag. Code running in the `if` constructs of `Meteor.isServer` will show results on the console. For the others, `Meteor.isClient` must be used and they can be evaluated in the browser on `http://localhost:8000`.
 
+## Coverage 
+Coverage can be found at `http://localhost:8000/coverage`. This will show the coverage for server files. To get the coverage for client,
+1. Open the console at `http://localhost:8000`.
+2. Run this command `Meteor.sendCoverage(function(stats,nbErr) {console.log(stats,nbErr);});`
```
Comment (brylie) [inline README.md]: Where do I run this command? Meteor shell?

**Comment by shailee-m** [comment]
> The test command runs and I get some failures. NOTE: I ran the tests twice and got the same failures.
> 
> ```
> I20200305-10:25:09.964(2)?   45 passing (28s)
> I20200305-10:25:09.964(2)?   9 failing
> I20200305-10:25:09.964(2)? 
> I20200305-10:25:09.965(2)?   1) Activities Aggregate Activities Should group by activity name and week  by default:
> I20200305-10:25:09.965(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> I20200305-10:25:09.965(2)?   
> I20200305-10:25:09.965(2)? 
> I20200305-10:25:09.965(2)?   2) Activities Aggregate Activities Should be able to group by activity name and start of month:
> I20200305-10:25:09.965(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> I20200305-10:25:09.965(2)?   
> I20200305-10:25:09.965(2)? 
> I20200305-10:25:09.965(2)?   3) Activities Aggregate Activities Should be able to group by facilitator name and start of week:
> I20200305-10:25:09.965(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> I20200305-10:25:09.966(2)?   
> I20200305-10:25:09.966(2)? 
> I20200305-10:25:09.966(2)?   4) Activities Aggregate Activities Should be able to group by facilitator name and start of month:
> I20200305-10:25:09.966(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> I20200305-10:25:09.966(2)?   
> I20200305-10:25:09.966(2)? 
> I20200305-10:25:09.966(2)?   5) Activities getAggregatedActivities Should group by activity name and week :
> I20200305-10:25:09.966(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> I20200305-10:25:09.966(2)?   
> I20200305-10:25:09.966(2)? 
> I20200305-10:25:09.966(2)?   6) Activities getAggregatedActivities Should be able to group by activity name and start of month:
> I20200305-10:25:09.967(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> I20200305-10:25:09.967(2)?   
> I20200305-10:25:09.967(2)? 
> I20200305-10:25:09.967(2)?   7) Activities getAggregatedActivities Should be able to group by facilitator name and start of week:
> I20200305-10:25:09.967(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> I20200305-10:25:09.967(2)?   
> I20200305-10:25:09.967(2)? 
> I20200305-10:25:09.967(2)?   8) Activities getAggregatedActivities Should be able to group by facilitator name and start of month:
> I20200305-10:25:09.967(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> I20200305-10:25:09.967(2)?   
> I20200305-10:25:09.967(2)? 
> I20200305-10:25:09.968(2)?   9) Activities getAllHomeReportAggregates Should return appropriate data for each aggregation on success:
> I20200305-10:25:09.968(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> ```

This is new. Usually, one or 2 random tests fail. Let me test on my end again.

File: README.md (hunk 1 above)
Comment (shailee-m) [reply README.md]: Go to http://localhost:8000. Open the developer-tools: console and type this command.

Comment (shailee-m) [reply README.md]: Browser console since it is for client tests

Comment (brylie) [reply README.md]: OK, please add those details to the README. I didn't figure that out :-)

**Comment by shailee-m** [comment]
> > The test command runs and I get some failures. NOTE: I ran the tests twice and got the same failures.
> > ```
> > I20200305-10:25:09.964(2)?   45 passing (28s)
> > I20200305-10:25:09.964(2)?   9 failing
> > I20200305-10:25:09.964(2)? 
> > I20200305-10:25:09.965(2)?   1) Activities Aggregate Activities Should group by activity name and week  by default:
> > I20200305-10:25:09.965(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> > I20200305-10:25:09.965(2)?   
> > I20200305-10:25:09.965(2)? 
> > I20200305-10:25:09.965(2)?   2) Activities Aggregate Activities Should be able to group by activity name and start of month:
> > I20200305-10:25:09.965(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> > I20200305-10:25:09.965(2)?   
> > I20200305-10:25:09.965(2)? 
> > I20200305-10:25:09.965(2)?   3) Activities Aggregate Activities Should be able to group by facilitator name and start of week:
> > I20200305-10:25:09.965(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> > I20200305-10:25:09.966(2)?   
> > I20200305-10:25:09.966(2)? 
> > I20200305-10:25:09.966(2)?   4) Activities Aggregate Activities Should be able to group by facilitator name and start of month:
> > I20200305-10:25:09.966(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> > I20200305-10:25:09.966(2)?   
> > I20200305-10:25:09.966(2)? 
> > I20200305-10:25:09.966(2)?   5) Activities getAggregatedActivities Should group by activity name and week :
> > I20200305-10:25:09.966(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> > I20200305-10:25:09.966(2)?   
> > I20200305-10:25:09.966(2)? 
> > I20200305-10:25:09.966(2)?   6) Activities getAggregatedActivities Should be able to group by activity name and start of month:
> > I20200305-10:25:09.967(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> > I20200305-10:25:09.967(2)?   
> > I20200305-10:25:09.967(2)? 
> > I20200305-10:25:09.967(2)?   7) Activities getAggregatedActivities Should be able to group by facilitator name and start of week:
> > I20200305-10:25:09.967(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> > I20200305-10:25:09.967(2)?   
> > I20200305-10:25:09.967(2)? 
> > I20200305-10:25:09.967(2)?   8) Activities getAggregatedActivities Should be able to group by facilitator name and start of month:
> > I20200305-10:25:09.967(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> > I20200305-10:25:09.967(2)?   
> > I20200305-10:25:09.967(2)? 
> > I20200305-10:25:09.968(2)?   9) Activities getAllHomeReportAggregates Should return appropriate data for each aggregation on success:
> > I20200305-10:25:09.968(2)?      Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
> > ```
> 
> This is new. Usually, one or 2 random tests fail. Let me test on my end again.

Here's what I do. I start the server. Once the process logs `listening on localhost:8000`, I change a file and save a file which triggers the watch and tests are rerun correctly. 

It fails the first time as the test start running before the server is up

**Comment by brylie** [comment]
> Here's what I do. I start the server. Once the process logs listening on localhost:8000, I change a file and save a file which triggers the watch and tests are rerun correctly.

OK, please add that to the documentation. Then, we can merge.

**Comment by shailee-m** [comment]
> > Here's what I do. I start the server. Once the process logs listening on localhost:8000, I change a file and save a file which triggers the watch and tests are rerun correctly.
> 
> OK, please add that to the documentation. Then, we can merge.

done

**Pull request merged**